### ISSUE TITLE
Geometry class is used by open-source edition, but docs say it is only supported in commercial editions

### ISSUE THREAD

**Issue by sgrimm** [jOOQ/jOOQ#13227]
### Expected behavior

The `Geometry` class shouldn't be documented as only available in commercial editions if it is used in the open-source edition.

OR

Generating code from a database that uses the PostgreSQL `GEOMETRY` type using the jOOQ 3.16 open-source edition shouldn't use the new `Geometry` class.

### Actual behavior

`GEOMETRY` columns are bound to the jOOQ `Geometry` class in Kotlin code that's generated by the open-source edition, despite the docs ([example](https://www.jooq.org/javadoc/3.16.5-SNAPSHOT/org.jooq/org/jooq/Geometry.html)) saying the class is only supported by the commercial editions.

### Steps to reproduce the problem

The problem may just be a documentation issue, but it doesn't require any custom configuration of the code generator; any table with a GEOMETRY column like

```
CREATE TABLE demo (location GEOMETRY);
```

will result in generated code that references the jOOQ `Geometry` class, even with the open-source edition of jOOQ 3.16:

```
    val LOCATION: TableField<DemoRecord, Geometry?> = createField(DSL.name("location"), SQLDataType.GEOMETRY, this, "")
```

### Versions

- jOOQ: 3.16.4
- Java: `openjdk version "17.0.2" 2022-01-18`
- Database (include vendor): PostgreSQL 14.1, PostGIS 3.1.4
- OS: OS X 12.2.1
- JDBC Driver (include name if inofficial driver): PostgreSQL 42.3.3

**Comment by lukaseder** [comment]
Thanks a lot for your report. This is going to be addressed soon, via https://github.com/jOOQ/jOOQ/issues/13143.

When jOOQ adds support for new out-of-the-box data types, it should be possible to opt out of this support and revert to the old behaviour of those types mapping to "unknown" `java.lang.Object`

The workaround would be to adapt your data type `Binding` to be a `Binding<Geometry, U>` instead of `Binding<Object, U>`.

Although this might not be intuitive, a `Binding<T, U>` doesn't *have to* use its own `Converter` (see https://github.com/jOOQ/jOOQ/issues/4819), and as such, the fact that the spatial extensions are commercial only starting from jOOQ 3.16, you shouldn't be limited by that.

Once https://github.com/jOOQ/jOOQ/issues/13143 is available, the third option will be implemented: The `Geometry` references are generated by default, to advertise the available feature from the commercial editions, but a hint about how to opt out of this will also be given. Does that make sense?

With the last rationale, I'm closing this as a duplicate of https://github.com/jOOQ/jOOQ/issues/13143

**Comment by lukaseder** [comment]
See also:
https://stackoverflow.com/q/71054033/521799

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/13143 has been implemented for jOOQ 3.17.0 and 3.16.5. The latter should be released this week. This will allow for opting out of recognising the `GEOMETRY` and `GEOGRAPHY` types in the code generator, treating them as `SQLDataType.OTHER`, as before.
